test/uk/gov/hmrc/traderservices/controllers/TimeFieldHelperSpec.scala

```diff
@@ -0,0 +1,157 @@
+      validate(("12", "00", "PM")) shouldBe Valid
+      validate(("", "", "")) shouldBe Invalid(ValidationError("error.foo.required"))
+      validate(("", "", "AM")) shouldBe Invalid(
+        Seq(ValidationError("error.foo.required-hour"), ValidationError("error.foo.required-minutes"))
+      )
+      validate(("12", "", "AM")) shouldBe Invalid(
+        Seq(ValidationError("error.foo.required-minutes"))
+      )
+      validate(("", "59", "AM")) shouldBe Invalid(
+        Seq(ValidationError("error.foo.required-hour"))
+      )
+      validate(("", "59", "")) shouldBe Invalid(
+        Seq(ValidationError("error.foo.required-hour"), ValidationError("error.foo.required-period"))
+      )
+      validate(("00", "", "SM")) shouldBe Invalid(
+        Seq(
+          ValidationError("error.foo.invalid-hour-value"),
+          ValidationError("error.foo.required-minutes"),
+          ValidationError("error.foo.invalid-period-value")
+        )
+      )
+      validate(("00", "60", "")) shouldBe Invalid(
+        Seq(
+          ValidationError("error.foo.invalid-hour-value"),
+          ValidationError("error.foo.invalid-minutes-value"),
+          ValidationError("error.foo.required-period")
+        )
+      )
+      validate(("01", "00", "M")) shouldBe Invalid(
+        Seq(
+          ValidationError("error.foo.invalid-period-value")
+        )
+      )
+      validate(("0a", "b0", "M")) shouldBe Invalid(
+        Seq(
+          ValidationError("error.foo.invalid-hour-digits"),
+          ValidationError("error.foo.invalid-minutes-digits"),
+          ValidationError("error.foo.invalid-period-value")
+        )
+      )
+    }
+
+    "validate and map time parts into a time" in {
+      timeFieldsMapping("bar").bind(Map("hour" -> "12", "minutes" -> "45", "period" -> "AM")) shouldBe Right(
+        LocalTime.parse("00:45")
+      )
+      timeFieldsMapping("bar").bind(Map("hour" -> "12", "minutes" -> "00", "period" -> "PM")) shouldBe Right(
+        LocalTime.parse("12:00")
+      )
+      timeFieldsMapping("bar").bind(Map("hour" -> "3", "minutes" -> "3", "period" -> "AM")) shouldBe Right(
+        LocalTime.parse("03:03")
+      )
+      timeFieldsMapping("bar").bind(Map("hour" -> "3", "minutes" -> "3", "period" -> "PM")) shouldBe Right(
+        LocalTime.parse("15:03")
+      )
+      timeFieldsMapping("bar").bind(Map()) should haveError("error.bar.required")
+      timeFieldsMapping("bar").bind(Map("hour" -> "", "minutes" -> "", "period" -> "")) should haveError(
+        "error.bar.required"
+      )
+      timeFieldsMapping("bar").bind(Map("hour" -> "", "minutes" -> "")) should haveError(
+        "error.bar.required"
+      )
+      timeFieldsMapping("bar").bind(Map("hour" -> "12")) should haveOnlyErrors(
+        "error.bar.required-minutes",
+        "error.bar.required-period"
+      )
+      timeFieldsMapping("bar").bind(Map("hour" -> "00", "minutes" -> "60")) should haveOnlyErrors(
+        "error.bar.invalid-hour-value",
```

Comment:
I know this is a draft so I assume the foos & bars will be updated - other than that seems to be fine - will be able to do more checking once page is accessible

Reply:
this is about testing generic mapping function, so foo and bar are here just as example